File: benchmark/ruby/insertion_sort.rb
start = Time.now.to_f * 1000
arr = (1..1000).to_a.reverse
(1...arr.length).each do |j|
    key = arr[j]
    k = j - 1
    while k >= 0 && arr[k] > key
        arr[k + 1] = arr[k]
        k -= 1
    end
    arr[k + 1] = key
end
finish = Time.now.to_f * 1000
puts "First: #{arr[0]}"
puts "Last: #{arr[-1]}"
puts "Time: #{(finish - start).round}ms"

File: benchmark/src/nodejs/insertion_sort.js
const arr = [];
for (let i = 1000; i > 0; i--) arr.push(i);
const start = Date.now();
for (let j = 1; j < arr.length; j++) {
    const key = arr[j];
    let k = j - 1;
    while (k >= 0 && arr[k] > key) {
        arr[k + 1] = arr[k];
        k--;
    }
    arr[k + 1] = key;
}
const end = Date.now();
console.log(`Result: ${arr[0]}/${arr[arr.length - 1]}`);
;
console.log(`Time: ${end - start}ms`);

